### PR TITLE
HPCC-19243 Build error in ICU with Ubuntu 18.04

### PR DESCRIPTION
**Pull request by richardkchapman**
Note - will need to set the new CMake variable ICU_REQUIRES_CPP11 to build. A
cleaner/more complete fix will likely be available for 7.0.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Compiled under 18.04, and can successfully compile a local program that includes unicode usage.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: rtl/eclrtl/CMakeLists.txt [hunk 1]
```diff
@@ -69,8 +69,8 @@ include_directories (
 ADD_DEFINITIONS( -D_USRDLL -DECLRTL_EXPORTS )
 
 if (USE_BOOST_REGEX)
-    if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
-        set_source_files_properties(eclregex.cpp PROPERTIES COMPILE_FLAGS -std=c++98)
+    if ((CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX) AND NOT ICU_REQUIRES_CPP11)
+     #   set_source_files_properties(eclregex.cpp PROPERTIES COMPILE_FLAGS -std=c++98)
```
Comment (ghalliday) [inline rtl/eclrtl/CMakeLists.txt]: I think commented out by mistake

Comment (richardkchapman) [reply rtl/eclrtl/CMakeLists.txt]: Repushed

**Comment by xwang2713** [comment]
Try to build https://github.com/richardkchapman/HPCC-Platform.git branch 1804a and get following erros:

1) 
HPCC-Platform/plugins/cassandra/cpp-driver/src/murmur3.cpp:118:17: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case 15: k2 ^= ((int64_t) (tail[14])) << 48;
              ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

2)
HPCC-Platform/plugins/cassandra/cpp-driver/src/buffer.hpp:199:16: error: ‘*((void*)(&<anonymous>)+8).cass::Buffer::data_.cass::Buffer::<unnamed union>::buffer’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
     RefBuffer* temp = data_.buffer;

3) 
HPCC-Platform/system/security/securesocket/securesocket.cpp:1655:5: error: ‘CRYPTO_mem_leaks’ was not declared in this scope
     CRYPTO_mem_leaks(bio_err);

**Comment by richardkchapman** [comment]
@xwang2713 Those are separate issues and should have separate Jiras raised for them.

**Comment by xwang2713** [comment]
OK

**Comment by richardkchapman** [comment]
The first two are probably already addressed in 7.0 and a simple backport of the CMakeLists.txt file will resolve - the last I don't know. I suspect it's a function that has been removed in newer openssl builds but someone that knows that code better than I do will probably have to fix.

**Comment by xwang2713** [comment]
I will try to build after merged and will open a new JIRA if build fails.

**Pull request merged**